src/contributing/pr_checklist.md

```diff
@@ -66,4 +66,45 @@ If new kinds of tests are added, please check the
 [testing](https://parallaxsecond.github.io/parsec-book/parsec_service/tests/index.html) page and its
 child.
 
+## Testing
+
+Pull requests introducing new features create new ways in which the service interacts with its
+environment. All such new avenues must be thoroughly tested to check both for potential flaws in the
+current implementation, and possible regressions due to future iterations or changes.
+
+A list of current types of tests can be found
+[here](https://parallaxsecond.github.io/parsec-book/parsec_service/tests/existing_tests.html). The
+list only looks at the high-level types of tests that we support, but should be a good guide to what
+types of tests must be enabled or written. For example:
+
+- When a **new provider** is introduced, or when support for some **new operations is added to an
+   existing provider**, make sure all the end-to-end tests that cover the new functionality are
+   enabled. This is generally done by means of Cargo features in the `e2e-tests` crate. If the
+   provider doesn't support the functionality needed for some test, make sure it is disabled using
+   such features, e.g. if only EC keys are supported for asymmetric signatures, tests based on RSA
+   keys should be explicitly disabled. If the provider includes functionality which does not have
```

Comment:
Maybe say how the test can be disabled? As in it's better to disable it at runtime (as check in the operation is supported or not) rather than not compiling it.

Reply:
Attempted a fix - does it sound more reasonable now?